### PR TITLE
Feature/test jest

### PR DESCRIPTION
**Pull request by adhikasp**
Add a sample unit test using [jest](https://jestjs.io/docs/en/) for:

- Formatter - partial
- WorkspaceNavigation component.

Natively, jest does not support ES6 style import `import F from 'utilities/formatter';`.
So I add babel compiler to preprocess js and vue file.

### CODE REVIEW

**Comment by adhikasp** [comment]
@ajthinking oh also, enabling CI integration for unit test and coverage would be cool :D

I think we can configure codeclimate or sonar for it.

File: src/defaultStore/state.js [hunk 1]
```diff
@@ -1,4 +1,9 @@
 export default function(options) {
+
+    if (typeof window.__ENV__ === 'undefined') {
```
Comment (adhikasp) [inline src/defaultStore/state.js]: Afterthought, this definition could be moved to test specific environment to avoid unnecessary change here.

**Comment by ajthinking** [comment]
This looks great, thanks man! I'll try to test it out and add some more cases later today, so might get back with a comment then. Merging for now. 

Yes CI is a must have :) Let me check up on codeclimate or sonar, I only used Travis CI before

**Comment by ajthinking** [comment]
Just implemented a few more tests, this is gonna be so much easier. Awesome 🚀🚀🚀
